seacrowd/sea_datasets/brcc/brcc.py

```diff
@@ -0,0 +1,111 @@
+  year      = {2022},
+  url       = {https://aclanthology.org/2022.coling-1.389/},
+  pages     = {4418--4428},
+}
+"""
+
+_LOCAL = False
+_LANGUAGES = ["zlm", "eng", "cmn"]
+_DATASETNAME = "brcc"
+_DESCRIPTION = """
+The Bahasa Rojak Crawled Corpus (BRCC) is a code-mixed dataset for the Bahasa Rojak dialect in Malaysia.
+Passages are generated through data augmentation from English and Malay Wikipedia pages using a modified CoSDA-ML method.
+The quality of generated passages is evaluated by two native Malay speakers.
+"""
+_HOMEPAGE = "https://data.depositar.io/dataset/brcc_and_sentibahasarojak"
+_LICENSE = Licenses.UNKNOWN.value
+_URL = "https://data.depositar.io/dataset/304d1572-27d6-4549-8292-b1c8f5e9c086/resource/8a558f64-98ff-4922-a751-0ce2ce8447bd/download/BahasaRojak_Datasets.zip"
+
+_SUPPORTED_TASKS = [Tasks.SELF_SUPERVISED_PRETRAINING]
+_SOURCE_VERSION = "1.0.0"
+_SEACROWD_VERSION = "1.0.0"
+
+
+class BRCCDataset(datasets.GeneratorBasedBuilder):
+    """Dataset of Bahasa Rojak passages generated from English and Malay Wikipedia pages."""
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    SEACROWD_VERSION = datasets.Version(_SEACROWD_VERSION)
+
+    BUILDER_CONFIGS = [
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_source",
+            version=SOURCE_VERSION,
+            description=f"{_DATASETNAME} source schema",
+            schema="source",
+            subset_id=_DATASETNAME,
+        ),
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_seacrowd_ssp",
+            version=SEACROWD_VERSION,
+            description=f"{_DATASETNAME} SEACrowd ssp schema",
+            schema="seacrowd_ssp",
+            subset_id=_DATASETNAME,
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = f"{_DATASETNAME}_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+        # Source schema = SeaCrowd schema because file only contains lines of text
+        if self.config.schema in ("source", "seacrowd_ssp"):
+            features = schemas.ssp_features
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: datasets.DownloadManager) -> List[datasets.SplitGenerator]:
+        """Returns SplitGenerators."""
+        data_dir = dl_manager.download_and_extract(_URL)
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={
+                    "filepath": os.path.join(data_dir, "BahasaRojak Datasets/BRCC/mix.train"),
+                    "split": "train",
+                },
+            )
+        ]
+
+    def _generate_examples(self, filepath: Path, split: str) -> Tuple[int, Dict]:
+        """Yields examples as (key, example) tuples."""
+        with open(filepath, encoding="utf-8") as f:
```

Comment:
Some optimization x)

![image](https://github.com/SEACrowd/seacrowd-datahub/assets/71018093/5a3950e1-4fc5-4d8a-9215-1b54562a7426)

Reply:
I saw the commit message "file line processing" but these lines did not change. Are you going to change this or keep it? I think it would be fine for both on general case :)

Reply:
The changes are in proceeding lines, let me know if you don't see it:

![image](https://github.com/SEACrowd/seacrowd-datahub/assets/77622465/e6499cc1-3cff-4357-b1b2-fbf2c2db99bb)

Reply:
OO yeah i see it now~
Confused haha, cause it wasnt marked as `Outdated` as well by Github. LGTM!